fsspec/implementations/http.py

```diff
@@ -253,6 +254,9 @@ async def _get_file(
         kw.update(kwargs)
         logger.debug(rpath)
         session = await self.set_session()
+        if await self._isdir(rpath):
```

Comment:
This will result in essentially every file getting downloaded twice, once to check the directory. This information should already be available in _get(), so we can skip there rather than here. It doesn't really make sense to call _get_file() on a path that is not considered a file.